Sources/TarkaUI/Components/NavigationRow/TUINavigationRow.swift

```diff
@@ -7,96 +7,102 @@
 
 import SwiftUI
 
-/// A view that displays a navigation row with an optional fluent icon and badge count.
+/// A view that displays a navigation row with an optional ImageIconProtocol icon and badge count.
 ///
-/// The navigation row is a horizontal stack with an optional fluent icon, title and an optional extra view.
+/// The navigation row is a horizontal stack with an optional ImageIconProtocol icon, title and an optional extra view.
 ///
 /// `minHeight` of this View is 40. This is to match the exact design of our Design System
 ///
 /// Example usage:
 ///
-///     TUINavigationRow(title: "Label")
+///     TUINavigationRow(title: "Label") { }
 ///
 /// - Parameters:
 ///   - title: The title to display in the navigation row.
+///   - action: The action to perform when the TUINavigationRow item is tapped.
+///
+/// - Returns: A closure that returns the content
 ///
 public struct TUINavigationRow: View {
-  var title: any StringProtocol
-  var leftIcon: ImageIconProtocol?
-  var isActive = false
-  var rightIcon: ImageIconProtocol?
-  var badgeCount: Int?
-  var showErrorBadge: Bool = false
+  
+  private var title: any StringProtocol
+  
+  private var inputStyle: InputStyle
+  
+  private var action: () -> Void
   
   /// Creates a navigation row with the specified title, symbol and an extra view.
   ///
   /// - Parameters:
   ///   - title: The title to display in the navigation row.
-  ///   - symbol: The symbol to display in the navigation row. The default value is `nil`.
-  ///   - accessoryView: Extra content to display in the navigation row.
+  ///   - action: The action to perform when the TUINavigationRow item is tapped.
   ///
-  public init(
-    title: any StringProtocol) {
-      self.title = title
-    }
+  public init(title: any StringProtocol, action: @escaping () -> Void) {
+    self.title = title
+    self.action = action
+    inputStyle = .init()
+  }
   
   public var body: some View {
-    
+    Button(action: action, label: mainView)
+      .buttonStyle(NavigationRowButtonStyle(for: inputStyle))
+      .accessibilityElement(children: .contain)
+      .accessibilityIdentifier(Accessibility.root)
```

Comment:
Could you check if the accessibility IDs are correct in the parent view wherever this component is used? You can also check with Gautham to see if the UI tests pass when running on this branch

Reply:
@arvindhsukumar Sure